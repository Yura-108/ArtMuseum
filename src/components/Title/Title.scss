@import '@styles/mixins';
@import '@styles/variables';

h1 {
  @include font-style('Lexend Deca', 700, 64px, 80px);
  text-transform: capitalize;
  color: $color-text-grey;
  div {
    @include flex-center;
    span {
      margin-left: 20px;
    }
  }
  span {
    color: $color-orange;
  }
  img {
    width: 64px;
  }
  @media (max-width: $width-main-container) {
    @include font-style('Lexend Deca', 500, 48px, 60px);
    img {
      width: 48px;
    }
  }

  @media (max-width: $width-main-container-small) {
    @include font-style('Lexend Deca', 500, 32px, 40px);
    span {
      margin-left: 5px;
    }
    img {
      width: 32px;
    }
  }
}
