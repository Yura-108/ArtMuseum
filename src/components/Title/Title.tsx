import './Title.scss';
import React from 'react';
<<<<<<< HEAD
import { TitleProps } from '@types/componentsPropsTypes.ts';

const Title: React.FC<TitleProps> = ({ children }) => {
=======
interface Props {
  children?: React.ReactNode | string;
}
const Title: React.FC<Props> = ({ children }) => {
>>>>>>> main
  return <h1>{children}</h1>;
};
export default Title;
