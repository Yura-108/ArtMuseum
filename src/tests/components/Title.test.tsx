import { render, screen } from '@testing-library/react';
import Title from '@components/Title/Title.tsx';
import '@testing-library/jest-dom';

describe('Title', () => {
  it('renders the Title component with children', () => {
    render(<Title>Test Title</Title>);

    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument();
    expect(screen.getByText('Test Title')).toBeInTheDocument();
  });

  it('renders correctly with different children', () => {
    render(<Title>Another Test</Title>);

    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument();
    expect(screen.getByText('Another Test')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Title>Snapshot Title</Title>);

    expect(asFragment()).toMatchSnapshot();
  });
});
