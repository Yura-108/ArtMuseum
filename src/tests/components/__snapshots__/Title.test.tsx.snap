// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    Snapshot Title
  </h1>
</DocumentFragment>
`;
